app/models/agents/digest_agent.rb

```diff
@@ -24,12 +26,18 @@ class DigestAgent < Agent
     def default_options
       {
           "expected_receive_period_in_days" => "2",
-          "message" => "{{ events | map: 'message' | join: ',' }}"
+          "message" => "{{ events | map: 'message' | join: ',' }}",
+          "clear_queue" => "0"
       }
     end
 
     form_configurable :message, type: :text
     form_configurable :expected_receive_period_in_days
+    form_configurable :clear_queue
+
+    def validate_options
+      errors.add(:base, 'clear_queue must be 0 to 999') unless options['clear_queue'].to_i >= 0 && options['clear_queue'].to_i < 1000
```

Comment:
The model for this change is Liquid Output Agent. As per its description, it caps the event limit at 1000 for "performance". I'll remove the limit if you think it is unnecessary.